src/main/java/org/cbioportal/cdd/repository/topbraid/TopBraidRepository.java

```diff
@@ -0,0 +1,108 @@
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ *
+ * @author Manda Wilson
+ **/
+@Repository
+public abstract class TopBraidRepository<T> {
+
+    private final static Logger logger = Logger.getLogger(TopBraidRepository.class);
+
+    @Value("${topbraid.url}")
+    private String topBraidURL;
+
+    @Autowired
+    private TopBraidSessionConfiguration topBraidSessionConfiguration;
+
+    protected List<T> query(String query, ParameterizedTypeReference<List<T>> parameterizedType)
+            throws TopBraidException {
+        return query(query, parameterizedType, true);
+    }
+
+    private List<T> query(String query, ParameterizedTypeReference<List<T>> parameterizedType, boolean refreshSessionOnFailure)
+            throws TopBraidException {
+        logger.debug("query() -- query: '" + query + "'");
+        String sessionId = topBraidSessionConfiguration.getSessionId();
+        logger.debug("query() -- sessionId: " + sessionId);
+        RestTemplate restTemplate = new RestTemplate();
+
+        // the default supported types for MappingJackson2HttpMessageConverter are:
+        //   application/json and application/*+json
+        // our response content type is application/sparql-results+json-simple
+        // NOTE: if the response content type was one of the default types we 
+        //   would not have to add the message converter to the rest template
+        MappingJackson2HttpMessageConverter messageConverter = new MappingJackson2HttpMessageConverter();
+        messageConverter.setSupportedMediaTypes(Collections.singletonList(
+            new MediaType("application","sparql-results+json-simple")));
+        restTemplate.getMessageConverters().add(messageConverter);
+
+        // set our JSESSIONID cookie and our params
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("Cookie", "JSESSIONID=" + sessionId);
+        MultiValueMap<String, String> map = new LinkedMultiValueMap<String, String>();
+        map.add("format", "json-simple");
+        map.add("query", query);
+        HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<MultiValueMap<String, String>>(map, headers);
+
+        // NOTE ParameterizedTypeReference cannot be made generic, that is why child class passes it
+        // See: http://stackoverflow.com/questions/21987295/using-spring-resttemplate-in-generic-method-with-generic-parameter
+        try {
+            ResponseEntity<List<T>> response = restTemplate.exchange(topBraidURL,
+                HttpMethod.POST,
+                request,
+                parameterizedType);
+            logger.debug("query() -- response.getBody(): '" + response.getBody() + "'");
```

Comment:
we do not check the status code of the response? Maybe it is enough that we don't see what we need in the body.

Reply:
@sheridancbio If you don't get what you want you will get an exception which is caught and handled.